gradle-plugins/src/main/java/com/linkedin/pegasus/gradle/PegasusPlugin.java

```diff
@@ -1390,6 +1445,40 @@ protected void configureRestModelGeneration(Project project, SourceSet sourceSet
     });
   }
 
+  protected void configurePegasusSchemaSnapshotGeneration(Project project, SourceSet sourceSet, boolean isExtensionSchema)
+  {
+    File schemaDir = isExtensionSchema? project.file(getExtensionSchemaPath(project, sourceSet))
+        : project.file(getDataSchemaPath(project, sourceSet));
+    if (isExtensionSchema && !SharedFileUtils.getSuffixedFiles(project, schemaDir, PDL_FILE_SUFFIX).isEmpty())
+    {
+      project.getLogger().debug("No extension schemas, skip generatePegasusExtensionSchemaSnapshot task");
+      return;
+    }
+    File publishablePegasusSchemaSnapshotDir = project.file(project.getBuildDir().getAbsolutePath()
+        + File.separatorChar + sourceSet.getName() + (isExtensionSchema ? PEGASUS_EXTENSION_SCHEMA_SNAPSHOT: PEGASUS_SCHEMA_SNAPSHOT));
+
+    Task generatePegasusSchemaSnapshot = generatePegasusSchemaSnapshot(project, sourceSet,
+        isExtensionSchema ? PEGASUS_EXTENSION_SCHEMA_SNAPSHOT: PEGASUS_SCHEMA_SNAPSHOT, schemaDir, publishablePegasusSchemaSnapshotDir);
+
+    Task checkSchemaSnapshot = project.getTasks().create(sourceSet.getTaskName("check",
+        isExtensionSchema ? PEGASUS_EXTENSION_SCHEMA_SNAPSHOT: PEGASUS_SCHEMA_SNAPSHOT),
+        CheckPegasusSnapshotTask.class, task ->
+        {
+          task.dependsOn(generatePegasusSchemaSnapshot);
+          // TODO: update CheckPegasusSnapshotTask
+        });
+
+    File pegasusSchemaSnapshotDir = new File(isExtensionSchema ? getPegasusExtensionSchemaSnapshotPath(project, sourceSet)
+        : getPegasusSchemaSnapshotPath(project, sourceSet));
+    pegasusSchemaSnapshotDir.mkdirs();
+
+    Task publishPegasusSchemaSnapshot = publishPegasusSchemaSnapshot(project, sourceSet,
+        isExtensionSchema ? PEGASUS_EXTENSION_SCHEMA_SNAPSHOT: PEGASUS_SCHEMA_SNAPSHOT, checkSchemaSnapshot,
+        publishablePegasusSchemaSnapshotDir, pegasusSchemaSnapshotDir);
+
+    project.getTasks().getByName(ASSEMBLE).dependsOn(publishPegasusSchemaSnapshot);
```

Comment:
In general, if support for Gradle 4 can be dropped, consider starting to use [task configuration avoidance](https://docs.gradle.org/current/userguide/task_configuration_avoidance.html). This involves using `TaskContainer#register` instead of `TaskContainer#create`, for example, and this line would become:

```
project.getTasks().named(ASSEMBLE,
    assembleTask -> assembleTask.dependsOn(publishPegasusSchemaSnapshot));
```

Feel free also to postpone that until a later time too, as it's not a completely trivial change.

Reply:
That's a good suggestion. I'd prefer to do that later. Thanks.